cpp/src/column/column.cu

```diff
@@ -85,6 +85,16 @@ column::contents column::release() noexcept
                           std::move(_children)};
 }
 
+std::size_t column::alloc_size() const
+{
+  return _data.size() + _null_mask.size() +
+         std::accumulate(
```

Comment:
Would it be more ergonomic to use `std::transform_reduce` here? https://en.cppreference.com/w/cpp/algorithm/transform_reduce

```cpp
std::transform_reduce(_children.begin(), _children.end(), std::size_t{0}, std::plus<>{},
    [](auto const& c){ return c->alloc_size(); }
);
```